src/promptix/core/components/layer_composer.py

```
@@ -0,0 +1,550 @@
+            return {}
+
+        try:
+            with open(config_path, encoding="utf-8") as f:
+                config = yaml.safe_load(f) or {}
+                self._config_cache[cache_key] = config
+                return config
+        except yaml.YAMLError as e:
+            raise ConfigurationError(
+                config_issue=f"Invalid YAML in config: {e}",
+                config_path=str(config_path)
+            )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)